### PR TITLE
[IMP] purchase_discount: load price with discount in stock_move when confirm a purchase order

### PR DESCRIPTION
**Pull request by agaldona**
Added new functionality, product unit price in stock move is taken with discount in order to correct the cost price calculus when it's method is average.

### CODE REVIEW

File: purchase_discount/models/purchase_order.py [hunk 1]
```diff
@@ -48,3 +48,13 @@ def _prepare_inv_line(self, account_id, order_line):
...
         result['discount'] = order_line.discount or 0.0
         return result
+
+    @api.model
+    def _prepare_order_line_move(self, order, order_line, picking_id,
+                                 group_id):
+        res = super(PurchaseOrder, self)._prepare_order_line_move(
+            order, order_line, picking_id, group_id)
+        for vals in res:
+            vals['price_unit'] = (vals.get('price_unit', 0.0) *
```
Comment (pedrobaeza) [inline purchase_discount/models/purchase_order.py:58]: Use directly `order_line.price_subtotal`

**Comment by pedrobaeza** [comment]
My doubt is what happens with this for dropshipping sales orders. I think the price is not correctly set. The sequence is:
- Create a sales order with dropshipping products (with supplierinfo added).
- Confirm the sales order
- A purchase order is generated
- When you confirm the purchase order, both delivery order and reception pickings are created from the same purchase order. The delivery order stock.move should have the sales prices (with or without discount, as defined in sales), not the purchase one.

Can you check it?

File: purchase_discount/models/purchase_order.py (hunk 1 above)
Comment (agaldona) [inline purchase_discount/models/purchase_order.py:58]: `price_subtotal` is calculated by quantity and this is a unit price

Comment (pedrobaeza) [inline purchase_discount/models/purchase_order.py:58]: OK, you're right

**Comment by agaldona** [comment]
When a dropshipping is done both delivery order and reception picking are one, there is only one picking created.
In that case the unit price must be taken from sale order, that means that the cost price will be calculated with the sale price.

**Comment by pedrobaeza** [comment]
Yeah, that's right. Thanks for seeing it.

:+1:

**Pull request merged**